include/tester/vtcp/VTCPServer.hpp

```diff
@@ -13,7 +13,8 @@ class VTCPServer : public TCPServer<VTCPSession>
     std::function<void()> startManualTest,
     std::function<bsoncxx::document::value()> getHistoryHeaders,
     std::function<bsoncxx::document::value(const std::string &)> getHistoryLog) :
-        TCPServer(port, startAutoTest, startManualTest, getHistoryHeaders, getHistoryLog)
+        TCPServer(std::move(port), std::move(startAutoTest), std::move(startManualTest),
```

Comment:
no need to use `std::move` on `port` which is a primitive `std::uint16_t`

Reply:
And remember - `TCPServer` is a generic class, unlike `VTCPServer` which is `VoIP-Tester` specific. There is no need to pass these function variables as arguments to the constructor of `TCPServer`.

Reply:
> no need to use `std::move` on `port` which is a primitive `std::uint16_t`

Done

Reply:
> And remember - `TCPServer` is a generic class, unlike `VTCPServer` which is `VoIP-Tester` specific. There is no need to pass these function variables as arguments to the constructor of `TCPServer`.

I need those functions in order to create `VTCPSession` in the class later on.

Reply:
I dislike the fact that now `TCPServer` is not generic because of those requirements - how about that instead of calling callbacks that you get through `std::bind` (specifically for these `VTCPSession` -> `Application` callbacks) you will use `Application::getInstance()`? that way, you won't need to have these callbacks in the `TCPServer` constructor. Just once you get a VTCP request for an opcode that requires the use of the `Application` class, you'll gain access to it using the `getInstance` method. That way this class remains generic.

Reply:
After we talked today - we agreed on not using the Singleton functionality of the `Application` class - you'll refactor the generic `TCP` relevant classes and the `VTCPServer` class to the new architecture we discussed.